### PR TITLE
ARTEMIS-671 Returning messages after connection killed, and validating usage of reconnect

### PR DESCRIPTION
**Pull request by clebertsuconic**
just for review for now... I'm running the full testsuite... will give an ok after it's done

### CODE REVIEW

File: artemis-commons/src/main/java/org/apache/activemq/artemis/logs/AssertionLoggerHandler.java [hunk 1]
```diff
@@ -71,6 +71,26 @@ public static boolean hasLevel(Level level) {
       return false;
    }
 
+   public static void dump() {
+      for (Map.Entry<String, ExtLogRecord> entry : messages.entrySet()) {
+         System.out.println(entry.getKey());
```
Comment (johnament) [inline artemis-commons/src/main/java/org/apache/activemq/artemis/logs/AssertionLoggerHandler.java]: Errr SOP?

Comment (clebertsuconic) [inline artemis-commons/src/main/java/org/apache/activemq/artemis/logs/AssertionLoggerHandler.java]: That's for debug only? 

What SOP means?

Comment (clebertsuconic) [inline artemis-commons/src/main/java/org/apache/activemq/artemis/logs/AssertionLoggerHandler.java]: Well.. it's useless anyway.. will remove it

**Comment by clebertsuconic** [comment]
I have finished the full testsuite run on this branch. a nice run.

If any objections.. this is good to be merged.

**Comment by johnament** [comment]
BTW -> SOP = System.out.println instead of logging.

**Comment by clebertsuconic** [comment]
@johnament I see.. it was intentionally SOP then... it would be on junit reports... it was from the test logger.. but it's been removed now.

**Pull request closed without merging**